functions/Get-DbaDbFileMap.ps1

```diff
@@ -0,0 +1,90 @@
+
+        For MFA support, please use Connect-DbaInstance.
+
+    .PARAMETER Database
+        The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+   .PARAMETER InputObject
+        Database object piped in from Get-DbaDatabase
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database
+        Author: Chrissy LeMaire (@cl), netnerds.net | Andreas Jordan (@JordanOrdix), ordix.de
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbFileMap
+
+    .EXAMPLE
+        PS C:\> $filemap = Get-DbaDbFileMap -SqlInstance sql2016 -Database test
+        PS C:\> Get-ChildItem \\nas\db\backups\test | Restore-DbaDatabase -SqlInstance sql2019 -Database test -FileMapping $filemap.FileMap
+
+        Restores test to sql2019 using the file structure built from the existing database on sql2016
+    #>
+    [CmdletBinding()]
+    param ([parameter(ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [object[]]$Database,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+        if (Test-Bound -not 'SqlInstance', 'InputObject') {
+            Write-Message -Level Warning -Message "You must specify either a SQL instance or supply an InputObject"
+            return
+        }
+
+        if (Test-Bound -Not -ParameterName InputObject) {
+            $InputObject += Get-DbaDatabase -SqlInstance $SqlInstance -SqlCredential $SqlCredential -Database $Database
+        }
+
+        foreach ($db in $InputObject) {
+            if ($db.IsAccessible) {
+                Write-Message -Level Verbose -Message "Processing database: $db"
+                $fileMap = @{ }
+
+                foreach ($file in $db.FileGroups.Files) {
+                    $fileMap[$file.Name] = $file.FileName
+                }
+                foreach ($file in $db.LogFiles) {
+                    $fileMap[$file.Name] = $file.FileName
+                }
+
+                [PSCustomObject]@{
+                    ComputerName = $db.ComputerName
+                    InstanceName = $db.InstanceName
+                    SqlInstance  = $db.SqlInstance
+                    Database     = $db.Name
+                    FileMap      = $fileMap
```

Comment:
Can we call this `FileMapping` since it will contain multiple files and it will match the parameter name we use in the Restore command.

Reply:
ohh i like it, yes

Reply:
One thing tied to this as well with the name of the command. Should we call it `Get-DbaDbFileMapping` as well? I mean when you think about how it is phrased when we are talking about it or maybe in docs as well, we term it has "you need to check the database mapping" or "what is the mapping of the database files" 🤷🏻

Reply:
One of those is "Mapping" plural for "Map" in this sense. If we are tying the command directly to the use of the parameter on the Restore command it would make sense to call the command the same as well. Not a hard yes for me, just thinking out loud.

I would also include an update to help on Restore for that parameter and give an example in that using this command so it is found more easily.

Reply:
Just renamed 👍🏼